src/client/components/Meal.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from "react";
+import { TiArrowBack } from "react-icons/ti"
+import { useParams } from "react-router-dom";
+import { useHistory } from "react-router-dom";
+import CheckAvailableReservations from "./CheckAvailableReservations";
+import ReviewForm from "./ReviewForm";
+import images from "./images";
+
+const Meal = () => {
```

Comment:
This component also does a lot of different things. In a small system it doesn't matter but in a larger project we want to make changes to only the relevant parts of the system. 
The title Meal component does not tell us very much. Some of your names do, like reReviewFormVisible. Good. But then you start to wonder what reviews are doing in the Meal component.